frontend/src/metabase/dashboard/context/context.tsx

```diff
@@ -282,14 +266,44 @@ const DashboardContextProviderInner = ({
     if (dashboardIdProp !== dashboardId) {
       fetchId(dashboardIdProp);
     }
-  }, [dashboardId, dispatch, fetchId, handleError, dashboardIdProp]);
+
+    async function fetchId(idToFetch: DashboardId) {
```

Comment:
Move inside the effect to limit the function scope since it's only used here anyway.